pages/_app.js

```diff
@@ -3,18 +3,41 @@ import { useState } from "react";
 import { initialShoppingList } from "@/lib/data";
 import { uid } from "uid";
 import sortShoppingListByCategory from "@/utils/sortShoppingListByCategory";
+import { useRouter } from "next/router";
 
 export default function App({ Component, pageProps }) {
   const [showForm, setShowForm] = useState(false);
   const [shoppingItems, setShoppingItems] = useState(initialShoppingList);
+  const [selectedItemId, setSelectedItemId] = useState(null);
 
   const handleAddItem = (newItem) => {
     const newTaskObject = { ...newItem, id: uid() };
     setShoppingItems([newTaskObject, ...shoppingItems]);
   };
 
+  const router = useRouter();
+
+  function handleDelete(id) {
+    setShoppingItems(shoppingItems.filter((task) => task.id !== id));
+    router.push("/");
```

Comment:
I assume you only want to close the popup because redirecting to another page is not necessary here. Using closeModal() / setShowForm should be enough here